downloads.markdown

```diff
@@ -16,45 +16,28 @@ This page describes the installation of the Haskell toolchain, which consists of
 
 *   [haskell-language-server](https://github.com/haskell/haskell-language-server) (optional): A language server for developers to integrate with their editor/IDE
 
-Select your platform to get more specific installation instructions:
+To install Haskell toolchain:
 
-*   [Linux, OS X and FreeBSD](#linux-mac-freebsd)
-
-*   [Windows](#windows)
-
-Alternative methods to install GHC are listed [here](#ghc-install-manual).
-
-* * *
-
-## Platform specific instructions
-
-### Linux, OS X and FreeBSD { #linux-mac-freebsd }
-
-1. Install GHC, cabal-install and haskell-language-server via [GHCup](https://www.haskell.org/ghcup/)
-2. To install stack, follow the instructions [here](https://docs.haskellstack.org/en/stable/install_and_upgrade/)
-
-*Note: Apple Silicon is not yet supported. You will have to install GHC in `arch -x86_64` compatibility mode using rosetta. See [here](https://derflounder.wordpress.com/2020/11/17/installing-rosetta-2-on-apple-silicon-macs/) and [here](https://gitlab.haskell.org/haskell/ghcup-hs/-/issues/101) for more pointers.*
-
-### Windows { #windows }
-
-1. Install GHC and cabal-install via Chocolatey
-	- [Configure Chocolatey](https://chocolatey.org/install) on your machine
-	- At an elevated command prompt, run `choco install haskell-dev`, followed by `refreshenv`.
-2. To install stack, follow the instructions [here](https://docs.haskellstack.org/en/stable/install_and_upgrade/#windows)
+1. Use [GHCup](https://www.haskell.org/ghcup/) to install 
```

Comment:
This suggests to install cabal/GHC via ghcup on windows and lists chocolatey as an alternative. Is that intentional?

Reply:
> Chocolatey was moved below under platform-specific instructions for package managers. If this move is controversial, I'm happy to restore it back, this was done in the interest of clarity and brevity, not because of partisan preferences.

True, I agree here. But the argument from @parsonsmatt was that neither windows nor stack support are sufficiently "battle-tested".

Reply:
Yeah this PR has the same problem as the other one, as far as I am concerned

I think the safest option is to add `ghcup` as an option under Windows, but not make it the preferred/recommended/"most obvious choice" until it's been in the wild for a bit more time.

Reply:
@parsonsmatt I'm sympathetic to your view in a context of the previous PR. However, here no installation method is removed, only one added. 

Note that ghcup is actually the **only** tool, fully capable of installing Haskell toolchain (as defined above). So unless there is a specific known issue with regards to Windows support in ghcup, I think it's worth to list it as proposed. But - again - I'm happy to leave it for haskell.org committee to deliberate and decide.

Reply:
This PR does quite a bit more than add an option to the list, which would be completely fine IMO.

The first presented option that is *relevant* to a user should be a well-tested "blessed" choice, with extremely little chance of failure or problems. The current page structure accomplishes this. The user selects their platform, and is given the 'blessed' path - `ghcup` for Linux/OSX/FreeBSD and `chocolatey`+`stack` for Windows.

The simplest change that adds `ghcup` as an option is to just add it as an option in the current Windows section.

```diff
 ### Windows { #windows }
 
 1. Install GHC and cabal-install via Chocolatey 
 	- [Configure Chocolatey](https://chocolatey.org/install) on your machine
 	- At an elevated command prompt, run `choco install haskell-dev`, followed by `refreshenv`.
 2. To install stack, follow the instructions [here](https://docs.haskellstack.org/en/stable/install_and_upgrade/#windows)
+3. `ghcup` recently acquired Windows support. To install `ghcup`, follow the instructions [here](wherever this needs to be)
```

This change accomplishes pretty much what everyone wants: `ghcup` gets mentioned under Windows support, `ghcup` gets exposure and more users and more testers, the `stack` team's concerns about `ghcup` as an installation method for `stack` are satisfied, my concerns about new-users being funneled into relatively less-tested workflows is abated. 

In ~6 months (or whatever timeline works for folks), when there is more confidence in `ghcup`'s Windows support, then we can switch to this unified installer plan. Then we have one main "How to get Haskell" section (heck yes) and alternatives for different platforms listed below.

Reply:
> This PR does quite a bit more than add an option to the list, which would be completely fine IMO.
> 
> The first presented option that is *relevant* to a user should be a well-tested "blessed" choice, with extremely little chance of failure or problems. The current page structure accomplishes this. The user selects their platform, and is given the 'blessed' path - `ghcup` for Linux/OSX/FreeBSD and `chocolatey`+`stack` for Windows.
> 
> The simplest change that adds `ghcup` as an option is to just add it as an option in the current Windows section.
> 
> ```diff
>  ### Windows { #windows }
>  
>  1. Install GHC and cabal-install via Chocolatey 
>  	- [Configure Chocolatey](https://chocolatey.org/install) on your machine
>  	- At an elevated command prompt, run `choco install haskell-dev`, followed by `refreshenv`.
>  2. To install stack, follow the instructions [here](https://docs.haskellstack.org/en/stable/install_and_upgrade/#windows)
> +3. `ghcup` recently acquired Windows support. To install `ghcup`, follow the instructions [here](wherever this needs to be)
> ```
> 
> This change accomplishes pretty much what everyone wants: `ghcup` gets mentioned under Windows support, `ghcup` gets exposure and more users and more testers, the `stack` team's concerns about `ghcup` as an installation method for `stack` are satisfied, my concerns about new-users being funneled into relatively less-tested workflows is abated. 
> 

This is specifically what I do not want: add more options/choice for the user.

The reason the DL page was confusing is exactly this: listing all sorts of installation methods, leaving the user confused.

Ultimately, it should be one single instruction. If that's not feasible, it should be one single instruction per platform. Alternatives are to be listed as a footnote or not at all.